packages/slate-editor/src/modules/editor/lib/insertUploadingFile.ts

```diff
@@ -29,17 +30,25 @@ export async function insertUploadingFile<T>(
     editor: Editor,
     {
         createElement,
-        ensureEmptyParagraphAfter,
         filePromise,
         loaderContentType,
         loaderMessage,
+        mode = 'replace',
     }: Parameters<T>,
 ): Promise<T | null> {
     const loader = createLoader({
         contentType: loaderContentType,
         message: loaderMessage,
     });
-    EditorCommands.insertNodes(editor, [loader], { ensureEmptyParagraphAfter });
+
+    if (mode === 'insert') {
+        EditorCommands.insertNodes(editor, [loader], { ensureEmptyParagraphAfter: true });
+    }
+
+    if (mode === 'replace') {
+        Transforms.setNodes(editor, loader);
```

Comment:
jfyi, as i remember setNodes will not set children, do not know if it affects something, but think it is worth to mention here